src/internet_identity/src/storage.rs

```diff
@@ -710,16 +735,27 @@ impl<M: Memory + Clone> Storage<M> {
         origin: &FrontendHostname,
     ) -> ApplicationNumber {
         let origin_hash = StorableOriginHash::from_origin(origin);
+        let origin_sha256 = StorableOriginSha256::from_origin(origin);
 
-        if let Some(existing_number) = self.lookup_application_with_origin_memory.get(&origin_hash)
+        // TODO[ID_352]: Switch to `lookup_application_with_origin_memory`.
+        if let Some(existing_number) = self
+            .lookup_application_with_origin_memory_old
+            .get(&origin_hash)
         {
             existing_number
         } else {
-            let new_number: ApplicationNumber = self.lookup_application_with_origin_memory.len();
+            let new_number: ApplicationNumber =
+                self.lookup_application_with_origin_memory_old.len();
 
-            self.lookup_application_with_origin_memory
+            // Update the source of truth.
+            // TODO[ID_353]: Remove this line.
+            self.lookup_application_with_origin_memory_old
                 .insert(origin_hash, new_number);
 
+            // Update the new map (will eventually become the source of truth).
+            self.lookup_application_with_origin_memory
+                .insert(origin_sha256, new_number);
```

Comment:
The new origin keeps the same identity numbers and if all goes well, after the migration checking the length or one will match the length of the other.

Should we add a check here with a log in case that's not the case? That could help us find a possible bug before fully moving to the new field.

Reply:
That's correct, I don't have. But I saw that there was still an open discussion with @sea-snake